doc/tutorial.md

```diff
@@ -337,6 +335,16 @@ seastar::future<> f() {
 }
 ```
 
+## Preemption and Task Quota
+
+As described above, an existing fiber of execution will yield back to the event loop when it performs a blocking operation such as IO or sleeping, as it has no more work to do until this blocking operation completes. Should a fiber have a lot of CPU bound work to do without any intervening blocking operations, however, it is important that execution is still yielded back to the event loop periodically.
+
+This is implemented via _preemption_: which can only occur at specific preemption points. At these points the fiber's remaining _task quota_ is checked and it has been exceeded the fiber yields. The task quota is a measure of how long tasks should be allowed to run before yielding to the event loop, and is set to 500 µs by default.
+
+It is important not to starve the event loop, as this would starve continuations of futures that weren't ready but have since become ready, and also starve the important **polling** done by the event loop (e.g., checking whether there is new activity on the network card). For example, iterating over a large container while doing CPU-bound work without any suspension points could starve the reactor and cause a _reactor stall_, which refers to a substantial period of time (e.g., more than 20 milliseconds) during which a task does not yield.
+
+Many seastar constructs such as looping constructs have built-in preemption points. You may also insert your own preemption points by calling `seastar::maybe_yield`, which performs a preemption check. Coroutines will also perform a preemption check at each `co_await`. Note that there is _not_ a preemption check between continuations attached to a future with `then()`, so a recursive future loop without explicit preemption checks may starve the reactor.
```

Comment:
There is no `seastar::maybe_yield` unless I'm missing something... There's seastar::thread::maybe_yield which only works inside a seastar::thread, and there's seastar::coroutine::maybe_yield()  (which you should use in a long loop in a coroutined function). For continuation-based code, you should use need_preempt() in a more elaborate manner (I'm not even sure what example to give for this).

Reply:
There is a vanilla version `seastar::maybe_yield` too, here: https://github.com/scylladb/seastar/blob/d7bd2e7178f55db4e5731b7a4168e22d509d7776/include/seastar/util/later.hh#L57

It works as you expect: effectively returning the `yield()` future if the task quota has expired, or a ready future otherwise.